src/app/App.tsx

```diff
@@ -1,11 +1,106 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import { TasksList } from 'pages/TaskList';
+import { CreateTask } from 'pages/CreateTask';
+
+import { Container, Area } from './App.styles';
+import { GlobalStyle } from 'styles/global';
+import { api } from 'services/api';
+import { ITasks } from 'types/tasks';
+
+import { v4 as uuidv4 } from 'uuid';
+
+interface Props {
+  title: string;
+  description: string;
+  isCompleted: boolean;
+}
 
 function App() {
+  const [tasks, setTasks] = useState<ITasks[]>([]);
+
+  useEffect(() => {
+    const loadTasksFromLocalStorage = () => {
+      const tasksJSON = localStorage.getItem('tasks');
+      return tasksJSON ? JSON.parse(tasksJSON) : [];
+    };
+
+    async function loadTodos() {
+      try {
+        const tasksData = await api();
+        setTasks(tasksData);
+      } catch (error) {
+        console.error('Error ao carregar lista de tarefas:', error);
+      }
+    }
+
+    const storedTasks = loadTasksFromLocalStorage();
+    if (storedTasks.length > 0) {
+      setTasks(storedTasks);
+    } else {
+      loadTodos();
+    }
+  }, []);
+
+  const saveTasksToLocalStorage = (tasks: ITasks[]) => {
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+  };
+
+  useEffect(() => {
+    saveTasksToLocalStorage(tasks);
+  }, [tasks]);
+
+  const handleAddNewTask = ({ title, description, isCompleted }: Props) => {
+    if (title.trim() && description.trim() !== '') {
+      const newTask = {
+        id: uuidv4(),
+        title: title,
+        description: description,
+        isCompleted: isCompleted,
+      };
+
+      setTasks([...tasks, newTask]);
```

Comment:
meio de setar estado poderia ser assim pra garantir pegar o estado antigo mais atualizado sem depender do estado atual:
```suggestion
      setTasks((oldState) => [...oldState , newTask]);
```
para saber mais: https://abidemi-dev.medium.com/react-hooks-usestate-with-previous-state-55b6272f1473